src/util/http.ts

```diff
@@ -21,24 +21,23 @@ export const getProxy = () => {
 
 export const getRequestOptions = (url: string) => {
     const parsedUrl = new URL(url);
+    let options;
 
-    const options = {
-        host: parsedUrl.hostname,
-        path: parsedUrl.pathname,
-        port: parsedUrl.port,
-        headers: { Host: '' }
-    };
-
-    if (getProxyVar() && options.host !== 'localhost' && options.host.substring(0, 3) !== '127') {
+    if (getProxyVar() && parsedUrl.hostname !== 'localhost' && parsedUrl.hostname.substring(0, 3) !== '127') {
+        options = <ClientRequestArgs>{};
         const proxy = getProxy();
+
         options.host = proxy.host;
         options.port = proxy.port;
         options.path = url;
-        options.headers.Host = parsedUrl.hostname;
+        options.headers = { 'Host': parsedUrl.hostname };
```

Comment:
did we mean to override the entire object here?

Reply:
That is correct. The headers object hasn't been created at this point. https://nodejs.org/api/http.html#http_http_request_options_callback